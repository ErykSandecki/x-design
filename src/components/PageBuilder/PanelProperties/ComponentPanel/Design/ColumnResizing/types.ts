export type TFocusElement = 'height' | 'width' | '';
