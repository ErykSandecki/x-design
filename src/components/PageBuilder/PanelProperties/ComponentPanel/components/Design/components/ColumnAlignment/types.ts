// types
import { AlignmentHorizontal, AlignmentVertical } from 'types';
import { TButtonGroup } from 'shared/UITools/ButtonGroup/types';

export type TAligmentHorizontalButtonGroup = Array<
  Pick<TButtonGroup, 'name' | 'tooltip'> & { key: AlignmentHorizontal }
>;
export type TAligmentVerticalButtonGoup = Array<Pick<TButtonGroup, 'name' | 'tooltip'> & { key: AlignmentVertical }>;
