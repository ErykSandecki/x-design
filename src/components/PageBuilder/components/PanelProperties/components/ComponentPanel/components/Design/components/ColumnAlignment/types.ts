// types
import { AlignmentHorizontal, AlignmentVertical } from 'types';
import { TButtonGroup } from 'shared/UITools/components/ButtonGroup/types';

export type TAligmentHorizontalButtonGroup = Array<Pick<TButtonGroup, 'name'> & { key: AlignmentHorizontal }>;
export type TAligmentVerticalButtonGoup = Array<Pick<TButtonGroup, 'name'> & { key: AlignmentVertical }>;
