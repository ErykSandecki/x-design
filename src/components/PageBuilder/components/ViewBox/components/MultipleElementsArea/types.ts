// types
import { TElement } from 'types';

export type TCoordinatesData = {
  elementsCoordinates: Array<{
    coordinates: TRectCoordinates;
    id: TElement['id'];
  }>;
  outline: TRectCoordinates;
};
