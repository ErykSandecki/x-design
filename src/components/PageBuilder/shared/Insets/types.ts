// types
import { TInsets } from 'types';

export type TInsetKeysGroup = [keyof Pick<TInsets, 'l' | 't'>, keyof Pick<TInsets, 'b' | 'r'>];
