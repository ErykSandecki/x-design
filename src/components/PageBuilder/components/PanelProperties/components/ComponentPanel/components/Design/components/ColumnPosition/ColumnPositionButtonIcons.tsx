import { ReactNode } from 'react';
import { useSelector } from 'react-redux';

// components
import ConstrainsView from '../../../../../../../../shared/ConstrainsView/ConstrainsView';

// store
import { dynamicDataSelector } from 'store/pageBuilder/selectors';

import { TSelectedElement } from 'store/pageBuilder/types';

const ColumnPositionButtonIcons = (firstElement: TSelectedElement, showConstrains: boolean): Array<ReactNode> => {
  const dynamicData = useSelector(dynamicDataSelector);

  return showConstrains ? [<ConstrainsView alignment={dynamicData[firstElement.id].alignment} key={0} />] : [];
};

export default ColumnPositionButtonIcons;
