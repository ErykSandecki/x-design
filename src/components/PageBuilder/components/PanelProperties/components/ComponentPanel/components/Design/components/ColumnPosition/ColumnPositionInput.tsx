import { ReactNode, useRef } from 'react';
import { useDispatch } from 'react-redux';

// components
import { ScrubbableInput, Small, TE2EValue, UITools } from 'shared';

// others
import { MAX, MIN } from '../../../../../../../../constants';

// store
import { updateEventsStatus } from 'store/pageBuilder/actions';

// types
import { ColorsTheme, KeyboardKeys } from 'types';

// utils
import { getValue } from './utils/getValue';
import { handleSubmitInput } from 'utils';

export type TColumnPositionInputProps = {
  disabled: boolean;
  disabledAll: boolean;
  e2eValue: TE2EValue;
  hasAlignment: boolean;
  isMultiple: boolean;
  label: string;
  onBlur: TFunc;
  onChange: TFunc<[string, boolean?]>;
  onMouseDown: TFunc;
  typeInput: HTMLInputElement['type'];
  value: string;
};

const ColumnPositionInput = ({
  disabled,
  disabledAll,
  e2eValue,
  hasAlignment,
  isMultiple,
  label,
  onBlur,
  onChange,
  onMouseDown,
  typeInput,
  value,
}: TColumnPositionInputProps): ReactNode => {
  const dispatch = useDispatch();
  const refInput = useRef<HTMLInputElement>(null);

  return (
    <UITools.TextField
      disabled={disabled}
      e2eValue={e2eValue}
      fullWidth
      onBlur={onBlur}
      onChange={(event) => onChange(event.target.value)}
      onClick={() => refInput.current.select()}
      onKeyDown={(event) => handleSubmitInput(KeyboardKeys.enter, refInput.current)(event)}
      ref={refInput}
      startAdornment={
        <ScrubbableInput
          disabled={disabled}
          e2eValue={e2eValue}
          max={MAX}
          min={MIN}
          onChange={(value) => onChange(value.toString(), true)}
          onMouseDown={onMouseDown}
          onMouseUp={() => dispatch(updateEventsStatus({ isMultipleMoving: false }))}
          value={isMultiple ? 0 : parseFloat(value)}
        >
          <Small color={ColorsTheme.neutral2}>{label}</Small>
        </ScrubbableInput>
      }
      type={typeInput}
      value={getValue(disabledAll, hasAlignment, isMultiple, value)}
    />
  );
};

export default ColumnPositionInput;
