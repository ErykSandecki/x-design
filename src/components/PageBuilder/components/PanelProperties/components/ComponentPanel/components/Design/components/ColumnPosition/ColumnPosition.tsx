import { FC } from 'react';
import { useTranslation } from 'react-i18next';

// components
import ColumnPositionButtonIcons from './ColumnPositionButtonIcons';
import ColumnPositionInput from './ColumnPositionInput';
import { UITools } from 'shared';

// hooks
import { usePositionEvents } from './hooks/usePositionEvents';

// others
import { translationNameSpace } from './constants';

// types
import { GridColumnType } from 'shared/UITools/components/Section/components/SectionColumn/enums';

// utils

const ColumnPosition: FC = () => {
  const { t } = useTranslation();

  const {
    disabledAll,
    disabledX,
    disabledY,
    firstElement,
    hasAlignmentHorizontal,
    hasAlignmentVertical,
    isMultiple,
    onBlurX,
    onBlurY,
    onChangeX,
    onChangeY,
    onMouseDown,
    showConstrains,
    typeInputX,
    typeInputY,
    x,
    y,
  } = usePositionEvents();

  return (
    <UITools.SectionColumn
      buttonsIcon={ColumnPositionButtonIcons(firstElement, showConstrains)}
      gridColumnType={GridColumnType.twoInputs}
      labels={[t(`${translationNameSpace}.label`)]}
      withMargin
    >
      <ColumnPositionInput
        disabled={disabledX}
        disabledAll={disabledAll}
        e2eValue="x"
        hasAlignment={hasAlignmentHorizontal}
        isMultiple={isMultiple}
        label="X"
        onBlur={onBlurX}
        onChange={onChangeX}
        onMouseDown={onMouseDown}
        typeInput={typeInputX}
        value={x}
      />
      <ColumnPositionInput
        disabled={disabledY}
        disabledAll={disabledAll}
        e2eValue="y"
        hasAlignment={hasAlignmentVertical}
        isMultiple={isMultiple}
        label="Y"
        onBlur={onBlurY}
        onChange={onChangeY}
        onMouseDown={onMouseDown}
        typeInput={typeInputY}
        value={y}
      />
    </UITools.SectionColumn>
  );
};

export default ColumnPosition;
