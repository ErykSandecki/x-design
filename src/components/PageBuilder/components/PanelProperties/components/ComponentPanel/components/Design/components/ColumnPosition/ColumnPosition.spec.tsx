import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';

// components
import ColumnPosition from './ColumnPosition';

// mocks
import {
  elementMock,
  elementDynamicDataMock,
  elementStaticDataMock,
  pageBuilderStateMock,
  selectedElementMock,
} from 'test/mocks/reducer/pageBuilderMock';

// others
import { REDUCER_KEY as PAGE_BUILDER } from 'store/pageBuilder/actionsType';

// store
import { configureStore } from 'store/store';

// types
import { AlignmentHorizontal, AlignmentVertical, E2EAttribute, KeyboardKeys } from 'types';

// utils
import { getByE2EAttribute } from 'test';

const currentPage = pageBuilderStateMock[PAGE_BUILDER].pages['0'];
const stateMock = {
  ...pageBuilderStateMock,
  [PAGE_BUILDER]: {
    ...pageBuilderStateMock[PAGE_BUILDER],
    pages: {
      ...pageBuilderStateMock[PAGE_BUILDER].pages,
      ['0']: {
        ...currentPage,
        elements: {
          allData: {
            ['-1']: {
              ...currentPage.elements.allData['-1'],
              children: [elementMock.id],
            },
            [elementMock.id]: {
              ...elementMock,
            },
          },
          dynamicData: {
            ['-1']: {
              ...currentPage.elements.dynamicData['-1'],
              children: [elementDynamicDataMock.id],
            },
            [elementDynamicDataMock.id]: elementDynamicDataMock,
          },
          staticData: {
            ['-1']: {
              ...currentPage.elements.staticData['-1'],
              children: [elementStaticDataMock.id],
            },
            [elementStaticDataMock.id]: {
              ...elementStaticDataMock,
            },
          },
        },
        selectedElements: [selectedElementMock],
      },
    },
  },
};

describe('ColumnPosition snapshots', () => {
  it('should render ColumnPosition', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ColumnPosition />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when values are mixed', () => {
    // mock
    const store = configureStore({
      ...stateMock,
      [PAGE_BUILDER]: {
        ...stateMock[PAGE_BUILDER],
        pages: {
          ['0']: {
            ...stateMock[PAGE_BUILDER].pages['0'],
            elements: {
              ...stateMock[PAGE_BUILDER].pages['0'].elements,
              allData: {
                ...stateMock[PAGE_BUILDER].pages['0'].elements.allData,
                ['-1']: {
                  ...stateMock[PAGE_BUILDER].pages['0'].elements.allData['-1'],
                  children: ['test-1', 'test-2'],
                },
                ['test-2']: {
                  ...elementMock,
                  coordinates: { x: 100, y: 100 },
                  id: 'test-2',
                },
              },
              dynamicData: {
                ...stateMock[PAGE_BUILDER].pages['0'].elements.dynamicData,
                ['-1']: {
                  ...stateMock[PAGE_BUILDER].pages['0'].elements.dynamicData['-1'],
                  children: ['test-1', 'test-2'],
                },
                ['test-2']: {
                  ...elementDynamicDataMock,
                  coordinates: { x: 100, y: 100 },
                  id: 'test-2',
                },
              },
              staticData: {
                ...stateMock[PAGE_BUILDER].pages['0'].elements.staticData,
                ['-1']: {
                  ...stateMock[PAGE_BUILDER].pages['0'].elements.staticData['-1'],
                  children: ['test-1', 'test-2'],
                },
                ['test-2']: {
                  ...elementStaticDataMock,
                  id: 'test-2',
                },
              },
            },
            selectedElements: [
              ...stateMock[PAGE_BUILDER].pages['0'].selectedElements,
              { ...selectedElementMock, id: 'test-2' },
            ],
          },
        },
      },
    });

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ColumnPosition />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when some element has alignment', () => {
    // mock
    const store = configureStore({
      ...stateMock,
      [PAGE_BUILDER]: {
        ...stateMock[PAGE_BUILDER],
        pages: {
          ['0']: {
            ...stateMock[PAGE_BUILDER].pages['0'],
            elements: {
              ...stateMock[PAGE_BUILDER].pages['0'].elements,
              allData: {
                ...stateMock[PAGE_BUILDER].pages['0'].elements.allData,
                ['-1']: {
                  ...stateMock[PAGE_BUILDER].pages['0'].elements.allData['-1'],
                  children: ['test-1', 'test-2'],
                },
                ['test-2']: {
                  ...elementMock,
                  alignment: {
                    horizontal: AlignmentHorizontal.center,
                    vertical: AlignmentVertical.center,
                  },
                  coordinates: { x: 100, y: 100 },
                  id: 'test-2',
                },
              },
              dynamicData: {
                ...stateMock[PAGE_BUILDER].pages['0'].elements.dynamicData,
                ['-1']: {
                  ...stateMock[PAGE_BUILDER].pages['0'].elements.dynamicData['-1'],
                  children: ['test-1', 'test-2'],
                },
                ['test-2']: {
                  ...elementDynamicDataMock,
                  alignment: {
                    horizontal: AlignmentHorizontal.center,
                    vertical: AlignmentVertical.center,
                  },
                  coordinates: { x: 100, y: 100 },
                  id: 'test-2',
                },
              },
              staticData: {
                ...stateMock[PAGE_BUILDER].pages['0'].elements.staticData,
                ['-1']: {
                  ...stateMock[PAGE_BUILDER].pages['0'].elements.staticData['-1'],
                  children: ['test-1', 'test-2'],
                },
                ['test-2']: {
                  ...elementStaticDataMock,
                  id: 'test-2',
                },
              },
            },
            selectedElements: [
              ...stateMock[PAGE_BUILDER].pages['0'].selectedElements,
              { ...selectedElementMock, id: 'test-2' },
            ],
          },
        },
      },
    });

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ColumnPosition />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('ColumnPosition behaviors', () => {
  it('should change positions X & Y when enter value on input', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { container } = render(
      <Provider store={store}>
        <ColumnPosition />
      </Provider>,
    );

    // find
    const inputX = getByE2EAttribute(container, E2EAttribute.textFieldInput, 'x');
    const inputY = getByE2EAttribute(container, E2EAttribute.textFieldInput, 'y');

    // action
    fireEvent.click(inputX);
    fireEvent.change(inputX, { target: { value: '100' } });
    fireEvent.keyDown(inputX, { key: KeyboardKeys.enter });
    fireEvent.blur(inputX);
    fireEvent.click(inputY);
    fireEvent.change(inputY, { target: { value: '100' } });
    fireEvent.keyDown(inputY, { key: KeyboardKeys.enter });
    fireEvent.blur(inputY);

    // result
    expect(store.getState()[PAGE_BUILDER].pages['0'].elements.allData['test-1'].coordinates).toStrictEqual({
      x: 100,
      y: 100,
    });
  });

  it('should change positions X & Y when triger ScrubbableInput', () => {
    // mock
    const store = configureStore(stateMock);
    const mouseMoveEvent = new MouseEvent('mousemove', {
      bubbles: true,
      cancelable: true,
      shiftKey: false,
      view: window,
    });
    Object.defineProperty(mouseMoveEvent, 'movementX', { value: -100 });

    // before
    const { container } = render(
      <Provider store={store}>
        <ColumnPosition />
      </Provider>,
    );

    // find
    const scrubbableInputX = getByE2EAttribute(container, E2EAttribute.scrubbableInput, 'x');
    const scrubbableInputY = getByE2EAttribute(container, E2EAttribute.scrubbableInput, 'y');

    // action
    fireEvent.mouseDown(scrubbableInputX, { clientX: 0, clientY: 0 });
    window.dispatchEvent(mouseMoveEvent);
    fireEvent.mouseUp(scrubbableInputX);
    fireEvent.mouseDown(scrubbableInputY, { clientX: 0, clientY: 0 });
    window.dispatchEvent(mouseMoveEvent);
    fireEvent.mouseUp(scrubbableInputY);

    // result
    expect(store.getState()[PAGE_BUILDER].pages['0'].elements.allData['test-1'].coordinates).toStrictEqual({
      x: -50,
      y: -50,
    });
  });
});
