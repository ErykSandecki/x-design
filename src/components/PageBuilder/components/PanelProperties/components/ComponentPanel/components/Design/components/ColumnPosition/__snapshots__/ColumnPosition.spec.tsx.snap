// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnPosition snapshots should render ColumnPosition 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t SectionColumn SectionColumn--light SectionColumn--with-margin SectionColumn--with-margin--light"
    data-test-section=""
  >
    <div
      class="SectionColumn__labels SectionColumn__labels--light"
      style="width: calc(100% - 24px - 0px - 8px);"
    >
      <small
        class="css-wcnq0t SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Position
      </small>
    </div>
    <div
      class="SectionColumn__wrapper SectionColumn__wrapper--light"
    >
      <div
        class="SectionColumn__content SectionColumn__content--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <div
          class="css-wcnq0t TextField TextField--light TextField--full-width TextField--full-width--light"
          data-test-text-field="x"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light"
            data-test-scrubbable-input="x"
          >
            <small
              class="css-wcnq0t Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              X
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="x"
            maxlength="6"
            type="number"
            value="0"
          />
        </div>
        <div
          class="css-wcnq0t TextField TextField--light TextField--full-width TextField--full-width--light"
          data-test-text-field="y"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light"
            data-test-scrubbable-input="y"
          >
            <small
              class="css-wcnq0t Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Y
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="y"
            maxlength="6"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="SectionColumn__buttons SectionColumn__buttons--light"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ColumnPosition snapshots should render when some element has alignment 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t SectionColumn SectionColumn--light SectionColumn--with-margin SectionColumn--with-margin--light"
    data-test-section=""
  >
    <div
      class="SectionColumn__labels SectionColumn__labels--light"
      style="width: calc(100% - 24px - 0px - 8px);"
    >
      <small
        class="css-wcnq0t SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Position
      </small>
    </div>
    <div
      class="SectionColumn__wrapper SectionColumn__wrapper--light"
    >
      <div
        class="SectionColumn__content SectionColumn__content--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <div
          class="css-wcnq0t TextField TextField--light TextField--disabled TextField--disabled--light TextField--full-width TextField--full-width--light"
          data-test-text-field="x"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light ScrubbableInput--disabled ScrubbableInput--disabled--light"
            data-test-scrubbable-input="x"
          >
            <small
              class="css-wcnq0t Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              X
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="x"
            disabled=""
            maxlength="6"
            type="text"
            value="Locked"
          />
        </div>
        <div
          class="css-wcnq0t TextField TextField--light TextField--disabled TextField--disabled--light TextField--full-width TextField--full-width--light"
          data-test-text-field="y"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light ScrubbableInput--disabled ScrubbableInput--disabled--light"
            data-test-scrubbable-input="y"
          >
            <small
              class="css-wcnq0t Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Y
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="y"
            disabled=""
            maxlength="6"
            type="text"
            value="Locked"
          />
        </div>
      </div>
      <div
        class="SectionColumn__buttons SectionColumn__buttons--light"
      >
        <div
          class="ConstrainsView ConstrainsView--light"
        >
          <div
            class="ConstrainsView__constrain ConstrainsView__constrain--light ConstrainsView__constrain--bottom ConstrainsView__constrain--bottom--light"
          />
          <div
            class="ConstrainsView__constrain ConstrainsView__constrain--light ConstrainsView__constrain--center-horizontal ConstrainsView__constrain--center-horizontal--light"
          />
          <div
            class="ConstrainsView__constrain ConstrainsView__constrain--light ConstrainsView__constrain--center-vertical ConstrainsView__constrain--center-vertical--light"
          />
          <div
            class="ConstrainsView__constrain ConstrainsView__constrain--light ConstrainsView__constrain--left ConstrainsView__constrain--left--light"
          />
          <div
            class="ConstrainsView__constrain ConstrainsView__constrain--light ConstrainsView__constrain--right ConstrainsView__constrain--right--light"
          />
          <div
            class="ConstrainsView__constrain ConstrainsView__constrain--light ConstrainsView__constrain--top ConstrainsView__constrain--top--light"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ColumnPosition snapshots should render when values are mixed 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t SectionColumn SectionColumn--light SectionColumn--with-margin SectionColumn--with-margin--light"
    data-test-section=""
  >
    <div
      class="SectionColumn__labels SectionColumn__labels--light"
      style="width: calc(100% - 24px - 0px - 8px);"
    >
      <small
        class="css-wcnq0t SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Position
      </small>
    </div>
    <div
      class="SectionColumn__wrapper SectionColumn__wrapper--light"
    >
      <div
        class="SectionColumn__content SectionColumn__content--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <div
          class="css-wcnq0t TextField TextField--light TextField--full-width TextField--full-width--light"
          data-test-text-field="x"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light"
            data-test-scrubbable-input="x"
          >
            <small
              class="css-wcnq0t Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              X
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="x"
            maxlength="6"
            type="text"
            value="Mixed"
          />
        </div>
        <div
          class="css-wcnq0t TextField TextField--light TextField--full-width TextField--full-width--light"
          data-test-text-field="y"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light"
            data-test-scrubbable-input="y"
          >
            <small
              class="css-wcnq0t Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Y
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="y"
            maxlength="6"
            type="text"
            value="Mixed"
          />
        </div>
      </div>
      <div
        class="SectionColumn__buttons SectionColumn__buttons--light"
      />
    </div>
  </div>
</DocumentFragment>
`;
